### PR TITLE
add a section about nested APIs in the tutorial

### PR DESCRIPTION
**Pull request by alpmestan**
This addresses #14. @jkarni Mind taking a look?

### CODE REVIEW

File: tutorial/server.md [hunk 1]
```diff
@@ -947,6 +947,186 @@ $ curl http://localhost:8081/foo
...
+  )
+
+-- we factor out a Header
+type API3 = Header "Authorization" Token :>
+  (    Get '[JSON] SecretData -- get some secret data, if authorized
+  :<|> ReqBody '[JSON] MoreSecretData :> Post '[] () -- add some secret data, if authorized
+  )
+```
+
+When taken to the extreme, so to speak, this approach lets you define APIs modularly and assemble them all into one big API type only at the end.
```
Comment (jkarni) [inline tutorial/server.md]: I'd remove `When taken to the extreme, so to speak,` and start with `This approach...`

File: tutorial/server.md [hunk 2]
```diff
@@ -947,6 +947,186 @@ $ curl http://localhost:8081/foo
...
+
+``` haskell
+type UserAPI1 = -- view the user with given userid, in JSON
+                Capture "userid" Int :> Get '[JSON] User
+          
+           :<|> -- delete the user with given userid. empty response
+                Capture "userid" Int :> Delete '[] ()
+```
+
+That's repetition, right here! Can't we somehow factor out the "userid", shared by both endpoints? As a matter of fact, we can.
```
Comment (jkarni) [inline tutorial/server.md]: Maybe just `We can instead factor out the "userid":`?

File: tutorial/server.md [hunk 3]
```diff
@@ -947,6 +947,186 @@ $ curl http://localhost:8081/foo
 not found
 ```
 
+## Nested APIs
+
+Let's take a break from introducing new combinators and functions and talk about how you can modularly define your *APIs*, avoiding as much as possible any kind of repetition. Consider this simple example:
```
Comment (jkarni) [inline tutorial/server.md]: Just "Let's see how you can define your APIs in a modular way, while avoiding repetition. Consider..."?

**Comment by jkarni** [comment]
Modulo my notes, LGTM. Sorry for having taken this long to respond!

**Comment by alpmestan** [comment]
Comments addressed. Feel free to merge when you're happy with the content.

**Pull request merged**